File: Projeto_2-Paulo Victor/src/seguro.py
from datetime import datetime

class Seguro:
    def __init__(self, carro, cliente, valor, vigencia):
        self.carro = carro
        self.cliente = cliente
        self.valor = valor
        self.vigencia = vigencia

    def calcular_valor(self, base_valor, taxa):
        self.valor = base_valor + (base_valor * taxa)
        return self.valor

    def verificar_vigencia(self):
        hoje = datetime.now()
        data_fim = datetime.strptime(self.vigencia.split(' a ')[1], "%d/%m/%Y")
        return hoje <= data_fim

File: Projeto_2/Projeto_2/src/seguro.js
class Seguro {
  constructor(carros, cliente, valor, vigencia) {
    this.carros = Array.isArray(carros) ? carros : [carros];
    this.cliente = cliente;
    this.valor = valor;
    this.vigencia = vigencia;
  }

  calcularValor(baseValor, taxa) {
    return baseValor + baseValor * taxa;
  }

  verificarVigencia() {
    const hoje = new Date();
    const dataFim = new Date(this.vigencia);
    return hoje <= dataFim;
  }

  exibirCarros() {
    return this.carros.map((carro) => carro.exibirDetalhes()).join("\n");
  }
}

export default Seguro;
